spec/models/book_spec.rb

```diff
@@ -21,6 +21,72 @@
 
       expect(book.errors[:unit_weight]).to include 'must be greater than or equal to 0'
     end
+
+    describe 'canonical model validations' do
+      let(:book) { build(:book, canonical_book:, game:) }
+      let(:game) { create(:game) }
+
+      context 'when the canonical model is not unique' do
+        let(:canonical_book) { create(:canonical_book) }
+
+        before do
+          create_list(
+            :book,
+            3,
+            canonical_book:,
+            game:,
+          )
+        end
+
+        it 'is valid' do
+          validate
+          expect(book.errors[:base]).to be_empty
+        end
+      end
+
+      context 'when the canonical model is unique' do
+        let(:canonical_book) do
+          create(
+            :canonical_book,
+            unique_item: true,
+            rare_item: true,
+          )
+        end
+
+        context 'when the canonical model has no books' do
+          before do
+            create(:book, canonical_book:)
```

Comment:
We shouldn't do this in a context called `'when the canonical model has no books'`.